magefile.go

```diff
@@ -652,6 +653,83 @@ func (t Test) IntegrationCIPrepare(ctx context.Context) error {
+		dev.WithKubeconfigPath(os.Getenv("KUBECONFIG")))
+	if err != nil {
+		return fmt.Errorf("creating cluster client: %w", err)
+	}
+
+	ctx = dev.ContextWithLogger(ctx, logger)
+	if err = injectStatusReportingEnvironmentVariable(ctx, cluster); err != nil {
+		return fmt.Errorf("Inject ENV into CSV failed: %w", err)
+	}
+	return nil
+}
+
+func injectStatusReportingEnvironmentVariable(ctx context.Context, cluster *dev.Cluster) error {
+	// OO_INSTALL_NAMESPACE is defined at:
+	// https://github.com/openshift/release/blob/master/ci-operator/config/openshift/addon-operator/openshift-addon-operator-main.yaml#L79
+	// The initial value("!create") is then over-written by the step https://steps.ci.openshift.org/reference/optional-operators-subscribe
+	// to the actual namespace string.
+	addonOperatorNS, found := os.LookupEnv("OO_INSTALL_NAMESPACE")
+	if found && addonOperatorNS != "!create" {
+		CSVList := &operatorsv1alpha1.ClusterServiceVersionList{}
+		err := cluster.CtrlClient.List(ctx, CSVList, ctrlclient.InNamespace(addonOperatorNS))
+		if err != nil {
+			return fmt.Errorf("listing csv's for patching error: %w", err)
+		}
+		if len(CSVList.Items) == 0 {
+			return fmt.Errorf("no csv's found in namespace: %s", addonOperatorNS)
+		}
+		addonOperatorCSV := CSVList.Items[0]
+		return patchAddonOperatorCSV(ctx, cluster, &addonOperatorCSV)
+	}
+	return fmt.Errorf("Invalid/Missing namespace value in OO_INSTALL_NAMESPACE: Got: %s", addonOperatorNS)
+}
+
+func patchAddonOperatorCSV(ctx context.Context,
+	cluster *dev.Cluster,
+	operatorCSV *operatorsv1alpha1.ClusterServiceVersion) error {
+	for i := range operatorCSV.Spec.InstallStrategy.StrategySpec.DeploymentSpecs {
+		currentDeployment := &operatorCSV.
+			Spec.
+			InstallStrategy.
+			StrategySpec.DeploymentSpecs[i]
+		// Find the addon operator deployment.
+		if currentDeployment.Name == "addon-operator-manager" {
+			for i := range currentDeployment.Spec.Template.Spec.Containers {
+				containerObj := &currentDeployment.Spec.Template.Spec.Containers[i]
+				// Find the addon operator manager container from the pod.
+				if containerObj.Name == "manager" {
+					if containerObj.Env == nil {
+						containerObj.Env = []corev1.EnvVar{}
+					}
+					// Set status reporting env variable to true.
+					containerObj.Env = append(containerObj.Env, corev1.EnvVar{
+						Name:  "ENABLE_STATUS_REPORTING",
+						Value: "true"},
+					)
+					break
+				}
+			}
+		}
+	}
+	if err := cluster.CtrlClient.Update(ctx, operatorCSV); err != nil {
+		return fmt.Errorf("patching ADO csv to inject env variable: %w", err)
+	}
+	// Wait for the newly patched deployment under the CSV to be ready.
+	deploymentObj := &appsv1.Deployment{}
+	deploymentObj.SetName("addon-operator-manager")
+	deploymentObj.SetNamespace(operatorCSV.Namespace)
+	if err := cluster.Waiter.WaitForReadiness(ctx, deploymentObj); err != nil {
+		return fmt.Errorf("waiting for addon operator deployment to be ready: %w", err)
+	}
```

Comment:
Considering OLM's flaky behaviour, imo it would be worth adding a check after this Readiness check to ensure that the `ENABLE_STATUS_REPORTING: "true"` did in fact propagate successfully to the `addon-operator-manager` deployment.